File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size()==0||nums.size()==1)
            return nums.size();
        sort(nums.begin(), nums.end());
        int count=1;
        int mc=1;
        for(int i=1;i<nums.size();i++)
        {
            int prev=nums[i-1];
            if(nums[i]-prev==1)
            {
                count++;
                mc=max(mc,count);
            }
            else if(nums[i]-prev>1)
            {
                count=1;
            }
            
        }
        return mc;
    }
};

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        n=len(nums)
        if n==0:return 0
        count=1
        ans=1
        prev=nums[0]
        nums.sort()
        for i in range(n):
            if nums[i]==1+prev:
                count+=1
            elif nums[i]!=prev:
                count=1
            prev=nums[i]
            ans=max(ans,count)
        return ans
            
        